go/extractor/cli/go-autobuilder/go-autobuilder.go

```diff
@@ -367,23 +375,50 @@ func getDepMode(emitDiagnostics bool) (DependencyInstallerMode, string) {
 	return GoGetNoModules, "."
 }
 
+type GoVersionInfo struct {
+	// The version string, if any
+	Version string
+	// A value indicating whether a version string was found
+	Found bool
+}
+
 // Tries to open `go.mod` and read a go directive, returning the version and whether it was found.
-func tryReadGoDirective(buildInfo BuildInfo) (string, bool) {
+func tryReadGoDirective(buildInfo BuildInfo) (GoVersionInfo, GoVersionInfo) {
+	noVersionInfo := GoVersionInfo{"", false}
+	compilerVersion := noVersionInfo
+	toolchainVersion := noVersionInfo
+
 	if buildInfo.DepMode == GoGetWithModules {
 		versionRe := regexp.MustCompile(`(?m)^go[ \t\r]+([0-9]+\.[0-9]+(\.[0-9]+)?)$`)
+		toolchainRe := regexp.MustCompile(`(?m)^toolchain[ \t\r]+([0-9]+\.[0-9]+(\.[0-9]+)?)$`)
```

Comment:
```suggestion
		toolchainRe := regexp.MustCompile(`(?m)^toolchain[ \t\r]+go([0-9]+\.[0-9]+(\.[0-9]+)?)$`)
```

Reply:
I'm also happy to remove the `$` from this and the line above to deal with things like `rc1` and `-suffix`.

Reply:
Done in https://github.com/github/codeql/pull/14194/commits/fe155d5d8bf346c009988f55f78b1935a2bc22df and https://github.com/github/codeql/pull/14194/commits/f009a5f1b0584f950dcb9e03ece586f0ae647f15